pkg/helper/versions/helpers.go

```diff
@@ -0,0 +1,127 @@
+package versions
+
+import (
+	"fmt"
+	"strings"
+
+	ver "github.com/hashicorp/go-version"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+func GetVersionList(r *rosa.Runtime, channelGroup string, isSTS bool, isHostedCP bool) (versionList []string,
+	err error) {
+	vs, err := r.OCMClient.GetVersions(channelGroup)
+	if err != nil {
+		err = fmt.Errorf("Failed to retrieve versions: %s", err)
+		return
+	}
+
+	for _, v := range vs {
+		if isSTS && !ocm.HasSTSSupport(v.RawID(), v.ChannelGroup()) {
+			continue
+		}
+		if isHostedCP {
+			valid, err := ocm.HasHostedCPSupport(v.RawID())
+			if err != nil {
+				return versionList, fmt.Errorf("failed to check HostedCP support: %v", err)
+			}
+			if !valid {
+				continue
+			}
+		}
+		versionList = append(versionList, v.RawID())
+	}
+
+	if len(versionList) == 0 {
+		err = fmt.Errorf("Could not find versions for the provided channel-group: '%s'", channelGroup)
+		return
+	}
+
+	return
+}
+
+func GetFilteredVersionList(versionList []string, minVersion string, maxVersion string) []string {
+	var filteredVersionList []string
+
+	// Parse the versions for comparison
+	min, errmin := ver.NewVersion(minVersion)
+	max, errmax := ver.NewVersion(maxVersion)
+
+	if errmin != nil || errmax != nil {
+		return versionList
+	}
+
+	for _, version := range versionList {
+		ver, errver := ver.NewVersion(version)
+		if errver != nil {
+			continue
+		}
+		if ver.GreaterThanOrEqual(min) && ver.LessThanOrEqual(max) {
+			filteredVersionList = append(filteredVersionList, version)
+		}
+	}
+	return filteredVersionList
+}
+
+// Used for hosted machinepool minimal version
+func GetMinimalHostedMachinePoolVersion(controlPlaneVersion string) (string, error) {
+	cpVersion, errcp := ver.NewVersion(controlPlaneVersion)
+	if errcp != nil {
+		return "", errcp
+	}
+	segments := cpVersion.Segments()
+	minor := segments[1] - 2
```

Comment:
Maybe we can add a comment here or use a const for `2`, just to help developers in the future.
I understand the calculation because I saw Guilherme's comment on the slack thread that "versions need to be within 2 minors from control plane".

Reply:
Totally +1, this code is cryptic and it needs both a constant and a proper comment explaining the logic.